samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -156,6 +161,42 @@ public void subscribeToJobModelVersionChange(IZkDataListener dataListener) {
   }
 
   /**
+   * publishes new job model into ZK
+   * @param jobModelVersion  version of the jobModeL to publish
+   * @param jobModel jobModel to publish
+   */
+  public void publishNewJobModel(String jobModelVersion, JobModel jobModel) {
+    try {
+      // This call should FAIL if the node already exists.
+      ObjectMapper mmapper = SamzaObjectMapper.getObjectMapper();
+      String jobModelStr = mmapper.writerWithDefaultPrettyPrinter().writeValueAsString(jobModel);
+      LOG.info("pid=" + processorId + " jobModelAsString=" + jobModelStr);
+      zkClient.createPersistent(keyBuilder.getJobModelPath(jobModelVersion), jobModelStr);
+      LOG.info("wrote jobModel path =" + keyBuilder.getJobModelPath(jobModelVersion));
+    } catch (Exception e) {
+      throw new SamzaException(e);
+    }
+  }
+
+  /**
+   * get the job model from ZK by version
+   * @param jobModelVersion jobModel version to get
+   * @return job model for this version
+   */
+  public JobModel getJobModel(String jobModelVersion) {
+    LOG.info("pid=" + processorId + "read the model ver=" + jobModelVersion + " from " + keyBuilder.getJobModelPath(jobModelVersion));
+    Object data = zkClient.readData(keyBuilder.getJobModelPath(jobModelVersion));
+    ObjectMapper mmapper = SamzaObjectMapper.getObjectMapper();
+    JobModel jm;
+    try {
+      jm = mmapper.readValue((String) data, JobModel.class);
+    } catch (IOException e) {
+      throw new SamzaException("failed to read JobModel from ZK", e);
+    }
+    return jm;
+  }
+
+  /**
    * read the jobmodel version from ZK
```

Comment:
Looks like getJobModelVersion() is used only in tests. Can you add @VisibleForTesting annotation?